### PR TITLE
Added logs to whoas using logback

### PR DESCRIPTION
**Pull request by maheshkelkar**
References #9

### CODE REVIEW

File: src/main/groovy/com/github/lookout/whoas/InMemoryQueue.groovy [hunk 1]
```diff
@@ -55,8 +59,12 @@ class InMemoryQueue extends AbstractHookQueue {
             action.call(item)
         }
         catch (Exception ex) {
+
             /* Put this back on the tail end of the queue */
+            logger.info("\"Pop\" on in-memory queue failed with an exception: " +
+                    ex.getMessage() + ", putting it back on tail-end")
```
Comment (rtyler) [inline src/main/groovy/com/github/lookout/whoas/InMemoryQueue.groovy]: Logback allows you to pass an exception object as the last argument and it will handle the proper printing for you automatically (tip from @rkuris).

This should be changed to:

```groovy
logger.info("\"Pop\" on in-memory queue failed, putting it back on the tail-end", ex)
```

File: src/main/groovy/com/github/lookout/whoas/Publisher.groovy [hunk 2]
```diff
@@ -41,18 +43,18 @@ class Publisher {
...
         try {
             response = inv.invoke()
-
-            /* LOG: response */
             String responseBody = response.readEntity(String.class)
         }
         catch (ProcessingException exc) {
-            /* LOG: warn on this exception */
+            logger.warn("\"POST\" to url: \"${request.url}\" " +
+                    "has failed with exception: " + exc.getMessage())
```
Comment (rtyler) [inline src/main/groovy/com/github/lookout/whoas/Publisher.groovy]: Same comment above about exception logging

File: src/main/groovy/com/github/lookout/whoas/RedisQueue.groovy [hunk 3]
```diff
@@ -120,6 +127,8 @@ class RedisQueue extends AbstractHookQueue {
                     action.call(request)
                 } catch (Exception ex) {
                     /* Put this back on the front of the queue */
+                    logger.info("\"Pop\" on redis queue failed with an exception: " +
+                            ex.getMessage() + ", pushing it back on front of the queue")
```
Comment (rtyler) [inline src/main/groovy/com/github/lookout/whoas/RedisQueue.groovy]: Same comment as above on exception handling

File: src/main/groovy/com/github/lookout/whoas/InMemoryQueue.groovy [hunk 4]
```diff
@@ -55,8 +59,11 @@ class InMemoryQueue extends AbstractHookQueue {
             action.call(item)
         }
         catch (Exception ex) {
+
             /* Put this back on the tail end of the queue */
+            logger.info("\"Pop\" on in-memory queue failed, putting it back on the tail-end", ex)
             this.internalQueue.put(item)
+            throw ex
```
Comment (rtyler) [inline src/main/groovy/com/github/lookout/whoas/InMemoryQueue.groovy]: I'm not sure that this rethrowing of the exception is correct (or the other one below in this pull request). Since these runners are likely going to be running inside of separate background thread, throwing these excveptions further up will likely result in crashing the calling thread won't it?

Comment (maheshkelkar) [inline src/main/groovy/com/github/lookout/whoas/InMemoryQueue.groovy]: * I was just trying to make this consistent with RedisQueue implementation. If we decide to change this, I will apply it equally in both places. We had added it so that it doesnt go unnoticed. (I was hitting this situation in my testing without any knowledge of what was happening)
* this code should get executed in the whoas (i.e. webhook thread) i.e. standalone thread. It should not have any impact on calling thread
* I dont think it will crash the program. I think some top level function will catch exception, log it and we will continue. But, now that we are logging execption using logback, we dont have to throw it again.
So in short, I can take this "throw" out from all exceptions.

**Comment by rtyler** [comment]
I merged this change out of band from GitHub to resolve some simple docs conflicts
